include/numerics/dense_vector.h

```diff
@@ -28,7 +28,7 @@
 
 #ifdef LIBMESH_HAVE_EIGEN
 #include "libmesh/ignore_warnings.h"
-#include <Eigen/Dense>
+#include <Eigen/Core>
 #include "libmesh/restore_warnings.h"
```

Comment:
This looks like a correct change, but probably belongs split into a separate PR to make bisecting easier if people encounter problems - surely including too much from Eigen wasn't causing --disable-amr problems?